src/main/java/io/strimzi/kafka/bridge/Application.java

```diff
@@ -132,25 +104,73 @@ private static Future<HttpBridge> deployHttpBridge(Vertx vertx, BridgeConfig bri
     }
 
+    private static Vertx createVertxInstance(BridgeConfig bridgeConfig) {
+        VertxOptions vertxOptions = new VertxOptions();
+        if (bridgeConfig.getMetrics() != null) {
+            vertxOptions.setMetricsOptions(metricsOptions()); // enable Vertx metrics
+        }
+        Vertx vertx = Vertx.vertx(vertxOptions);
+        return vertx;
+    }
+
     /**
-     * Return a JmxCollectorRegistry instance with the YAML configuration filters
+     * Set up the Vert.x metrics options
      *
-     * @return JmxCollectorRegistry instance
-     * @throws MalformedObjectNameException
-     * @throws IOException
+     * @return instance of the MicrometerMetricsOptions on Vert.x
      */
-    private static JmxCollectorRegistry getJmxCollectorRegistry()
+    private static MicrometerMetricsOptions metricsOptions() {
+        Set<String> set = new HashSet<>();
+        set.add(MetricsDomain.NAMED_POOLS.name());
+        set.add(MetricsDomain.VERTICLES.name());
+        return new MicrometerMetricsOptions()
+            .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true))
+            // define the labels on the HTTP server related metrics
+            .setLabels(EnumSet.of(Label.HTTP_PATH, Label.HTTP_METHOD, Label.HTTP_CODE))
+            // disable metrics about pool and verticles
+            .setDisabledMetricsCategories(set)
+            .setJvmMetricsEnabled(true)
+            .setEnabled(true);
+    }
+
+    private static MetricsReporter getMetricsReporter(BridgeConfig bridgeConfig) 
             throws MalformedObjectNameException, IOException {
-        InputStream is = Application.class.getClassLoader().getResourceAsStream("jmx_metrics_config.yaml");
-        if (is == null) {
-            return null;
+        if (bridgeConfig.getMetrics() != null) {
+            if (bridgeConfig.getMetrics().equals(MetricsType.JMX_EXPORTER.toString())) {
+                return new MetricsReporter(getJmxCollectorRegistry(bridgeConfig));
+            } else if (bridgeConfig.getMetrics().equals(MetricsType.STRIMZI_REPORTER.toString())) {
+                return new MetricsReporter(new StrimziCollectorRegistry());
+            }
         }
+        return null;
+    }
 
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
-            String yaml = reader
+    /**
+     * Return a JmxCollectorRegistry instance with the YAML configuration filters.
+     * This is loaded from a custom config file if present, or from the default configuration file.
+     *
+     * @return JmxCollectorRegistry instance
+     * @throws MalformedObjectNameException
+     * @throws IOException
+     */
+    private static JmxCollectorRegistry getJmxCollectorRegistry(BridgeConfig bridgeConfig) throws MalformedObjectNameException, IOException {
+        if (bridgeConfig.getJmxExporterConfigPath() != null && Files.exists(bridgeConfig.getJmxExporterConfigPath())) {
```

Comment:
If the file does not exist, you should probably exit with an error and not silently ignore it.

Reply:
I agree, we should log an error and fail if the configured file is not found. Otherwise we risk the user never realizing that it is not running with the custom config simply due to a small typo, for example. Fixed.

Reply:
I don't understand why this cannot cause exit. There is a different between the config file path not being set and being set to an non-existing file. I don't think you have to treat it the same way for both cases.